File: lintcode/422_length-of-last-word/length-of-last-word.py
# coding:utf-8
'''
@Copyright:LintCode
@Author:   chenweikang
@Problem:  http://www.lintcode.com/problem/length-of-last-word
@Language: Python
@Datetime: 16-04-24 18:07
'''

class Solution:
    # @param {string} s A string
    # @return {int} the length of last word
    def lengthOfLastWord(self, s):
        lst = s.split()
        return 0 if not lst else len(lst[-1])

File: lintcode/422_length-of-last-word/length-of-last-word.cpp
/*
@Copyright:LintCode
@Author:   chenweikang
@Problem:  http://www.lintcode.com/problem/length-of-last-word
@Language: C++
@Datetime: 16-04-24 18:02
*/

class Solution {
public:
    /**
     * @param s A string
     * @return the length of last word
     */
    int lengthOfLastWord(string& s) {
        stringstream ss(s);
        string one;
        int res = 0;
        while (ss >> one) res = one.size();
        return res;
    }
};